In [20]:
import pandas as pd

items_df = pd.read_csv("items_cleaned.csv")


In [21]:
items_df.info()
items_df.describe()
items_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 470356 entries, 0 to 470355
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Unnamed: 0       470356 non-null  int64  
 1   Restaurant_Name  470356 non-null  object 
 2   Menu_Name        470356 non-null  object 
 3   Dish_Name        470356 non-null  object 
 4   Dish_order       470356 non-null  int64  
 5   Dish_Price       470356 non-null  float64
dtypes: float64(1), int64(2), object(3)
memory usage: 21.5+ MB


Unnamed: 0                                    Restaurant_Name Menu_Name  \
0           0  Circle K - Bia Lạnh Các Loại - 5-4A Trung Hòa ...     SNACK   
1           1  Circle K - Bia Lạnh Các Loại - 5-4A Trung Hòa ...     SNACK   
2           2  Circle K - Bia Lạnh Các Loại - 5-4A Trung Hòa ...     SNACK   
3           3  Circle K - Bia Lạnh Các Loại - 5-4A Trung Hòa ...     SNACK   
4           4  Circle K - Bia Lạnh Các Loại - 5-4A Trung Hòa ...     SNACK   

                                           Dish_Name  Dish_order  Dish_Price  
0  Bánh Que Vị Tôm Dorkbua 55G (1 Gói)-8850157500371          10     19000.0  
1  Khoai Tây Chiên Vị Kim Chi O'Star 108G (1 Gói)...          10     25000.0  
2      Bánh Vị Cua Kinh Đô 32G (1 Gói)-8934680012652          10      7000.0  
3  Swing Khoai Tây Chiên Vị Bò BíT TếT 63G(1 ...           9     14000.0  
4  OISHI Snack Bắp Ngọt 73g (1 Gói)-8934803026078           8     13000.0

In [22]:
items_df["Dish_Price"].describe()

count    4.703560e+05
mean     1.121041e+05
std      2.150365e+05
min      1.000000e+00
25%      2.700000e+04
50%      4.900000e+04
75%      1.100000e+05
max      2.450000e+07
Name: Dish_Price, dtype: float64

In [23]:
top_dishes = items_df["Dish_Name"].value_counts().head(10)
top_dishes


Dish_Name
Coca               442
Trà chanh          410
Khoai tây chiên    338
Nem chua rán       277
Pepsi              265
Khoai lang kén     258
Trà đào cam sả     254
Trà Chanh          243
Nước ép dưa hấu    242
Trà đào            241
Name: count, dtype: int64

In [27]:
import plotly.express as px

# Đếm tần suất món ăn và lấy Top 10
top_dishes = items_df["Dish_Name"].value_counts().head(10)

# Chuyển sang DataFrame để dễ vẽ
top_dishes_df = top_dishes.reset_index()
top_dishes_df.columns = ["Dish_Name", "Count"]

# Vẽ biểu đồ thanh dọc
fig = px.bar(
    top_dishes_df,
    x="Dish_Name",
    y="Count",
    text="Count",
    title="Top 10 Món Ăn Phổ Biến Nhất",
    labels={"Dish_Name": "Tên Món Ăn", "Count": "Số Lần Xuất Hiện"}
)

# Tuỳ chỉnh hiển thị
fig.update_traces(
    texttemplate='%{text}',
    hovertemplate='Món: %{x}<br>Số lần xuất hiện: %{y}',
    marker_color='seagreen'
)

fig.update_layout(
    title_font=dict(size=22, family='Arial'),
    title_x=0.5,
    xaxis_title="Tên Món Ăn",
    yaxis_title="Số Lần Xuất Hiện",
    xaxis_tickangle=-45,  # xoay nhãn để không bị chồng lên nhau
    height=500
)

fig.show()


Danh sách Top 10 món phổ biến nhất cho thấy phần lớn là đồ uống và đồ ăn nhẹ, trong đó Coca (442 lần) và Trà chanh (410 lần) là hai món được bán nhiều nhất. Các món ăn vặt như Khoai tây chiên, Nem chua rán, Khoai lang kén cũng xuất hiện với tần suất cao. Ngoài ra, các loại trà trái cây như Trà đào cam sả, Trà đào, Trà chanh chiếm số lượng lớn, cho thấy thị hiếu tiêu dùng nghiêng về các loại đồ uống giải khát.
Những món này thường dễ chế biến, giá hợp lý, và phù hợp với học sinh, sinh viên – nhóm khách hàng chính trên các nền tảng giao đồ ăn như ShopeeFood.

In [24]:
# Món đắt nhất
most_expensive = items_df.sort_values(by="Dish_Price", ascending=False).head(10)
most_expensive[["Dish_Name", "Dish_Price", "Restaurant_Name"]]

Dish_Name  Dish_Price  \
63704   Yến sào huyết nguyên chiếc cao cấp Khánh Hòa 100g  24500000.0   
63707                  Yến sào hồng sơ chế Khánh Hòa 100g  19250000.0   
63701             Yến sào trắng sơ chế Khánh Hòa 100g TP1  10450000.0   
310927                       Cam sành hôm nay hơi bị ngon  10000000.0   
63705              Tổ yến hồng nguyên chiếc Khánh Hòa 50g   9840000.0   
317071                              Thịt bò Wagyu thăn A5   9500000.0   
72391                                   Saffron salam 20g   9000000.0   
63698             Yến xào trắng sơ chế Khánh Hòa 100g TP2   8999000.0   
63697             Yến sào trắng sơ chế Khánh Hòa 100g TP3   8250000.0   
63696             Yến xào trắng sơ chế Khánh Hòa 100g TP4   7700000.0   

                                          Restaurant_Name  
63704     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng  
63707     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng  
63701     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng  
310927  Tiệm Bé Cam - Nước Ép Tươi Nguyên Chất & Nước ...  
63705     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng  
317071               HANOFARM Thực Phẩm Sạch - Times City  
72391                Saffron Việt Nam Nhuỵ Hoa - An Dương  
63698     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng  
63697     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng  
63696     Yến Sạch Yến Nhà - Yến Rút Lông - Trần Duy Hưng

In [26]:
import plotly.express as px

# Đọc dữ liệu đã lọc sạch
items_df = pd.read_csv("items_cleaned.csv")

# Lấy Top 5 món đắt nhất
top10_expensive = items_df.sort_values(by="Dish_Price", ascending=False).head(10)

# Vẽ biểu đồ thanh ngang
fig = px.bar(
    top10_expensive[::-1],  # đảo ngược để món đắt nhất nằm trên cùng
    x="Dish_Price",
    y="Dish_Name",
    orientation="h",
    text="Dish_Price",
    title="Top 5 Món Ăn Đắt Nhất",
    labels={"Dish_Price": "Giá (VNĐ)", "Dish_Name": "Tên Món"}
)

# Tuỳ chỉnh hiển thị
fig.update_traces(
    texttemplate='%{text:,.0f}đ',
    hovertemplate='Món: %{y}<br>Giá: %{x:,.0f}đ',
    marker_color='crimson'
)

fig.update_layout(
    title_font=dict(size=22, color='black'),
    title_x=0.5,
    height=400,
    xaxis_title="Giá món ăn (VNĐ)",
    yaxis_title="Tên món ăn"
)

fig.show()